org.eclipse.scanning.device.ui/src/org/eclipse/scanning/device/ui/Activator.java

```diff
@@ -79,5 +88,22 @@ public ImageData getImageData() {
 		return imageDescriptorFromPlugin(PLUGIN_ID, path);
 	}
 
+	@Override
+	public <T> T getService(Class<T> serviceClass) {
+		if (context==null) return null;
+		ServiceReference<T> ref = context.getServiceReference(serviceClass);
+		return context.getService(ref);
+	}
+
+	@Override
+	public <T> Collection<T> getServices(Class<T> serviceClass) throws InvalidSyntaxException {
+		if (context==null) return null;
+		Collection<ServiceReference<T>> refs = context.getServiceReferences(serviceClass, null);
+		if (refs==null) return null;
+		Collection<T> ret = new LinkedHashSet<T>(refs.size());

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace the type specification in this constructor call with the diamond operator ("<>"). (sonar.java.source not set. Assuming 7 or greater.) [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2293)